src/ast.ts

```diff
@@ -302,31 +302,12 @@ function joinShouldIncludeCompilations(...compilations: string[]) {
   // remove empty strings
   let filteredCompilations = compilations.filter(it => it);
 
-  // FIXME(boopathi): Maybe, find a better fix?
-  //
-  // remove "true" since we are joining with '&&' as `true && X` = `X`
-  // This prevents an explosion of `&& true` which could break
-  // V8's internal size limit for string.
-  //
-  // Note: the `true` appears if a field does not have a skip/include directive
-  // So, the more nested the query is, the more of unnecessary `&& true`
-  // we get.
-  //
-  // Failing to do this results in [RangeError: invalid array length]
-  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Invalid_array_length
-  filteredCompilations = filteredCompilations.filter(
-    it => it.trim() !== "true"
+  // Split conditions by && and flatten it
+  filteredCompilations = ([] as string[]).concat(
+    ...filteredCompilations.map(e => e.split(" && "))
```

Comment:
let us disregard empty spaces, if any, in this deduplication.

```suggestion
    ...filteredCompilations.map(e => e.split(" && ").map(it => it.trim()))
```

Reply:
Should we have a test for this?